src/infill.cpp

```diff
@@ -42,6 +43,58 @@ static inline int computeScanSegmentIdx(int x, int line_width)
 
 namespace cura {
 
+void Infill::generate(VariableWidthPaths& toolpaths, Polygons& result_polygons,  Polygons& result_lines, const SierpinskiFillProvider* cross_fill_provider, const SliceMeshStorage* mesh)
+{
+    // generate walls
+    if (outer_contour.empty())
+    {
+        return;
+    }
+    if (wall_line_count > 0)
+    {
+        WallToolPaths wall_toolpaths(outer_contour, infill_line_width, wall_line_count, mesh->settings);
+        toolpaths = wall_toolpaths.generate();
+    }
+    outline_offset -= wall_line_count * infill_line_width;
+
+    if (infill_multiplier > 1)
+    {
+        bool zig_zaggify_real = zig_zaggify;
+        if (infill_multiplier % 2 == 0)
+        {
+            zig_zaggify = false;
+        }
+        Polygons generated_result_polygons;
+        Polygons generated_result_lines;
+        _generate(generated_result_polygons, generated_result_lines, cross_fill_provider, mesh);
+        zig_zaggify = zig_zaggify_real;
+        multiplyInfill(generated_result_polygons, generated_result_lines);
+        result_polygons.add(generated_result_polygons);
+        result_lines.add(generated_result_lines);
+    }
+    else
+    {
+        //_generate may clear() the generated_result_lines, but this is an output variable that may contain data before we start.
+        //So make sure we provide it with a Polygons that is safe to clear and only add stuff to result_lines.
+        Polygons generated_result_polygons;
+        Polygons generated_result_lines;
+        _generate(generated_result_polygons, generated_result_lines, cross_fill_provider, mesh);
+        result_polygons.add(generated_result_polygons);
+        result_lines.add(generated_result_lines);
+    }
+    if (connect_polygons)
+    {
+        // remove too small polygons
+        coord_t snap_distance = infill_line_width * 2; // polygons with a span of max 1 * nozzle_size are too small
+        auto it = std::remove_if(result_polygons.begin(), result_polygons.end(), [snap_distance](PolygonRef poly) { return poly.shorterThan(snap_distance); });
+        result_polygons.erase(it, result_polygons.end());
+
+        PolygonConnector connector(infill_line_width, infill_line_width * 3 / 2);
```

Comment:
Maybe add some brackets? This made me look up the operator precedence again ;-)

Reply:
Now you're getting the hang of the 'maximalist' approach, since I didn't touch this ;-)
But agreed I will change it :-)